src/Linux/Host/Broker/Impl/Services/LinuxSystemInfoService.cs

```diff
@@ -111,8 +118,103 @@ public class LinuxSystemInfoService : ISystemInfoService {
+
+            string productPart = "product: ";
+            string vendorPart = "vendor: ";
+            string busPart = "bus info: ";
+
+            // Match anything that looks like 1024M or 1024MB
+            Regex pattern = new Regex("[0-9]+[GM][B]?", RegexOptions.IgnoreCase);
+
+            int count = 1;
+            for (int i=0;  i < lshwDisplayData.Length; ++i) {
+                // find the line that starts with "*-" 
+                if (lshwDisplayData[i].StartsWithIgnoreCase("*-")) {
+                    string productName = GetPart(lshwDisplayData, productPart, ref i);
+                    string vendorName = GetPart(lshwDisplayData, vendorPart, ref i);
+                    string busInfo = GetPart(lshwDisplayData, busPart, ref i);
+                    long vram = 0;
+                    if (!string.IsNullOrEmpty(busInfo)) {
+                        int index = busInfo.IndexOf('@');
+                        if (index >= 0 && index < busInfo.Length) {
+                            string busvalue = busInfo.Substring(index + 1);
+                            for (int j = 0; j < dmesgData.Length; ++j) {
+                                if (dmesgData[i].ContainsIgnoreCase(busvalue)) {
+                                    var match = pattern.Match(dmesgData[i]);
+                                    if (match.Success) {
+                                        vram = GetRamValueMB(match.Value);
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    _videoCardInfo.Add(new VideoCardInfo() { VideoCardName = productName, VideoRAM = vram, VideoProcessor = vendorName });
+                    ++count;
+                }
+            }
+        }
+
+        private static long GetRamValueMB(string vramStr) {
+            long ram = 0;
+            Regex numPattern = new Regex("[0-9]+");
+            var match = numPattern.Match(vramStr);
+            
+            if (match.Success) {
+                long.TryParse(match.Value, out ram);
+            }
+
+            if (vramStr.ContainsIgnoreCase("G")) {
+                ram *= 1024;
+            }
+
+            return ram;
+        }
+
+        private IEnumerable<string> ExecuteAndGetOutput(string command, string arguments) {
+            Process proc = null;
+            List<string> standardOutData = new List<string>();
+            try {
+                ProcessStartInfo psi = new ProcessStartInfo();
+                psi.Arguments = arguments;
+                psi.FileName = command;
+                psi.RedirectStandardInput = true;
+                psi.RedirectStandardOutput = true;
+                psi.RedirectStandardError = true;
+                proc = _ps.Start(psi);
+
+                using (StreamReader reader = new StreamReader(proc.StandardOutput.BaseStream)) {
```

Comment:
`proc.StandardOutput` is already a reader - why do we need to create another one here?

Also, it seems that it would be easier to just do `ReadToEnd` to read everything at once?